venom panic on failure

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x100af2860]

goroutine 1 [running]:
github.com/ovh/venom.(*TestStepResult).appendError(0x0, {0x101fb3618?, 0x140004a6540?})
	/Users/fsamin/Documents/src/github.com/ovh/venom/types.go:204 +0x60
github.com/ovh/venom.(*Venom).runTestSteps(0x1400018cc00, {0x101fbdce8, 0x140005aee70}, 0x1400038b7c0, 0x0)
	/Users/fsamin/Documents/src/github.com/ovh/venom/process_testcase.go:196 +0x66c
github.com/ovh/venom.(*Venom).runTestCase(0x1018c0d28?, {0x101fbdce8, 0x140005aec00}, 0x1400029e2d0, 0x1400038b7c0)
	/Users/fsamin/Documents/src/github.com/ovh/venom/process_testcase.go:150 +0x528
github.com/ovh/venom.(*Venom).runTestCases(0x1400018cc00, {0x101fbdce8, 0x140005aec00}, 0x1400029e2d0)
	/Users/fsamin/Documents/src/github.com/ovh/venom/process_testsuite.go:122 +0x2b0
github.com/ovh/venom.(*Venom).runTestSuite(0x1400018cc00, {0x101fbdcb0, 0x102b50680}, 0x1400029e2d0)
	/Users/fsamin/Documents/src/github.com/ovh/venom/process_testsuite.go:73 +0xc44
github.com/ovh/venom.(*Venom).Process(0x1400018cc00, {0x101fbdcb0, 0x102b50680}, {0x140003a4980?, 0x8?, 0x14000299a80?})
	/Users/fsamin/Documents/src/github.com/ovh/venom/process.go:185 +0x178
github.com/ovh/venom/cmd/venom/run.glob..func2(0x1400018c200?, {0x140003a4980?, 0x4?, 0x101651943?})
	/Users/fsamin/Documents/src/github.com/ovh/venom/cmd/venom/run/cmd.go:400 +0x884
github.com/spf13/cobra.(*Command).execute(0x1029896e0, {0x140003a4960, 0x2, 0x2})
	/Users/fsamin/Documents/src/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x658
github.com/spf13/cobra.(*Command).ExecuteC(0x14000452000)
	/Users/fsamin/Documents/src/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x320
github.com/spf13/cobra.(*Command).Execute(0x140000726e8?)
	/Users/fsamin/Documents/src/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968 +0x1c
main.main()
	/Users/fsamin/Documents/src/github.com/ovh/venom/cmd/venom/main.go:12 +0x5c

--- a/process_testcase.go
+++ b/process_testcase.go
@@ -193,7 +193,9 @@ func (v *Venom) runTestSteps(ctx context.Context, tc *TestCase, tsIn *TestStepRe
 		ranged, err := parseRanged(ctx, rawStep, stepVars)
 		if err != nil {
 			Error(ctx, "unable to parse \"range\" attribute: %v", err)
-			tsIn.appendError(err)
+			testStepResult := TestStepResult{}
+			testStepResult.appendError(err)
+			tc.TestStepResults = append(tc.TestStepResults, testStepResult)
 			return
 		}
 

